xenvm: Enable BT USB

DKlimenko opened the pull request:
The V4H Evaluation Module does not have BT module.
Enable BT USB config in kernel and add Tp-Link UB500 BT USB support in kernel bluetooth drivers

DKlimenko commented:
Hi @lorc , @arminn  could you please review? Thanks in advance.

lorc commented on arch/arm64/configs/android_xenvm_defconfig:
```diff
@@ -149,7 +149,7 @@ CONFIG_IKCONFIG_PROC=y
 CONFIG_MEMCG=y
 CONFIG_MEMCG_SWAP=y
-CONFIG_RT_GROUP_SCHED=y
+# CONFIG_RT_GROUP_SCHED is not set
```
How realtime scheduling is related to BT USB in kernel?

DKlimenko replied:
Created separate commit for CONFIG_RT_GROUP_SCHED

lorc reviewed (changes requested):
> Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

What is this? I don't remember giving you my R-b.

lorc reviewed (changes requested):
> Tested-by: Dmytro Klimenko <dmytro_klimenko@epam.com>

BTW, this means nothing. You expected to test it, as you authored it. Only Tested-by by other people is relevant here.

DKlimenko committed:
xenvm: Enable real time scheduling for Android BT.

With enabled CONFIG_RT_GROUP_SCHED config,
Android BT crashed with following messages:
F droid.bluetooth: [0611/193616.263814:FATAL:btu_task.cc(125)]
main_thread_start_up: unable to enable real time scheduling

Disable CONFIG_RT_GROUP_SCHED config
for correct Android BT stack functionality.

Signed-off-by: Dmytro Klimenko <dmytro_klimenko@epam.com>
Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

DKlimenko committed:
xenvm: enable BT USB in kernel configuration

The V4H Evaluation Module does not have BT module.
Enabled BT USB support in kernel configuration.

Signed-off-by: Dmytro Klimenko <dmytro_klimenko@epam.com>
Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

DKlimenko committed:
drivers: Add Tp-Link UB500 BT USB support

Extend supported BT USB devices with Tp-Link UB500 (RTL8761B chip).

Signed-off-by: Dmytro Klimenko <dmytro_klimenko@epam.com>
Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>

lorc commented:
Please fix checkpatch warnings. As it was mentioned in https://xen-troops.github.io/process/process.html, which I asked you to read carefully.

```
lorc@hp:linux/ (6ade5755824c0✗) $ scripts/checkpatch.pl -g HEAD~3..HEAD                                                                                                             [15:54:12]
--------------------------------------------------------------------------
Commit 36a32606c9ba ("xenvm: Enable real time scheduling for Android BT.")
--------------------------------------------------------------------------
WARNING: Possible unwrapped commit description (prefer a maximum 75 chars per line)
#6: 
With enabled CONFIG_RT_GROUP_SCHED config, Android BT crashed with following messages:

total: 0 errors, 1 warnings, 8 lines checked

NOTE: For some of the reported defects, checkpatch may be able to
      mechanically convert to the typical style using --fix or --fix-inplace.

Commit 36a32606c9ba ("xenvm: Enable real time scheduling for Android BT.") has style problems, please review.
--------------------------------------------------------------------
Commit 027c8f03f2f3 ("xenvm: enable BT USB in kernel configuration")
--------------------------------------------------------------------
total: 0 errors, 0 warnings, 9 lines checked

Commit 027c8f03f2f3 ("xenvm: enable BT USB in kernel configuration") has no obvious style problems and is ready for submission.
-----------------------------------------------------------------
Commit 6ade5755824c ("drivers: Add Tp-Link UB500 BT USB support")
-----------------------------------------------------------------
total: 0 errors, 0 warnings, 9 lines checked

Commit 6ade5755824c ("drivers: Add Tp-Link UB500 BT USB support") has no obvious style problems and is ready for submission.

NOTE: If any of the errors are false positives, please report
      them to the maintainer, see CHECKPATCH in MAINTAINERS.
```

DKlimenko commented:
All checkpatch warnings were fixed.

lorc reviewed (approved):
Thank you

`Reviewed-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

DKlimenko commented:
Thanks for support.
